pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -63,7 +72,21 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 	}
 	totalNumNodes := len(nodeInfos)
 
-	score := calculatePriority(sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes), len(pod.Spec.Containers))
+	c, err := state.Read(preScoreStateKey)
```

Comment:
In general, you should assume that PreScore might be disabled, in which case you should run the logic to get all the images in Score.

Reply:
https://github.com/kubernetes/kubernetes/blob/06cf52262796f386260beab3bfb4d0a29ab7b665/pkg/scheduler/framework/plugins/imagelocality/image_locality.go#L77-L78
I didn't see that other plugin do that, but they should. I only need to do hack with creating empty node list, but it isn't need in preScore function anyway. It is here only because of the interface.

Reply:
Can you follow a similar approach? That is, have a unexported method that just knows how to get the data without writing it to the state.